File: src/main/java/org/grandviewtech/entity/enums/CoilType.java
package org.grandviewtech.entity.enums;

/*
 * #%L
 * Programmable Login Controller Inteface
 * %%
 * Copyright (C) 2016 GrandViewTech
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */

public enum CoilType
	{
	LABEL("LABEL"), END("END"), JUMP("JUMP"), OUTPUT("OUTPUT"), LINE("LINE"), LOAD("LOAD"), ROUTINE("ROUTINE"), LEFT_LINK("LEFT_LINK"), RIGHT_LINK("RIGHT_LINK"), PARALLEL("PARALLEL"), DEFAULT("DEFAULT"), DELETE("DELETE");
		private String coilType;
		
		private CoilType(String coilType)
			{
				this.coilType = coilType;
			}
			
		public String getCoilType()
			{
				return coilType;
			}
	}

File: ladder-core/crates/ladder-model/src/coil_type.rs
use serde::{Deserialize, Serialize};

/// Ported from org.grandviewtech.entity.enums.CoilType.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum CoilType {
    Label,
    End,
    Jump,
    Output,
    Line,
    Load,
    Routine,
    LeftLink,
    RightLink,
    Parallel,
    Default,
    Delete,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn wire_form_matches_original_java_enum_strings() {
        assert_eq!(serde_json::to_string(&CoilType::Load).unwrap(), "\"LOAD\"");
        assert_eq!(serde_json::to_string(&CoilType::LeftLink).unwrap(), "\"LEFT_LINK\"");
        assert_eq!(serde_json::to_string(&CoilType::RightLink).unwrap(), "\"RIGHT_LINK\"");
        assert_eq!(serde_json::to_string(&CoilType::Routine).unwrap(), "\"ROUTINE\"");
    }

    #[test]
    fn round_trips_through_json() {
        let value: CoilType = serde_json::from_str("\"END\"").unwrap();
        assert_eq!(value, CoilType::End);
    }
}
